### ISSUE TITLE
Regent: Issue with returning regentlib.arrays

### ISSUE THREAD

**Issue by ReadingSteiner122** [StanfordLegion/legion#1400]
I have run into an issue with returning `regentlib.arrays`.

```
task returnRArray()
    var test : regentlib.array(int, 1)
    test[0] = 0.0
    return test
end

task main()
   var test_receive : regentlib.array(int, 1)
   test_receive = returnRArray()
   format.println("{}", test_receive[0])
end
regentlib.start(main)
```
Here is the error that is thrown:
```
python3 ../legion/language/regent.py regentlibarray.rg 
/home/legion/language/terra/src/terralib.lua:1202: Errors reported during typechecking.
/home/legion/language/src/regent/codegen.t:1057: expected an array or pointer but found regent_array(int32, 1)
      `([value_expr.value][ [index.value] ]))
```
This issue disappears if I inline `returnRArray()`. This issue is not reproduced with integer arrays.

Please let me know if anything else is required.

**Comment by magnatelee** [comment]
Should be fixed by [49a3ad38](https://gitlab.com/StanfordLegion/legion/-/commit/49a3ad384c53d9382c0b4cd213ba17895b591eff). Please pull and try again.

**Comment by magnatelee** [comment]
@ReadingSteiner122 did you have a chance to confirm the fix?
